src/http-client.ts

```diff
@@ -0,0 +1,78 @@
+import request from 'superagent';
+
+import reqlogger from './request-logger';
+import { RetryPromise, retryPromise } from './utils';
+
+export type RequestApiOptions = {
+  method?: string;
+  data?: any;
+  host: string;
+  version?: string;
+  contentType?: string;
+  authToken?: string;
+  logger?: any
+};
+
+export class HttpClient {
+  private _apiRequestsMap = new Map<string, RetryPromise<any>>();
+
+  static retryStatusCodes = new Set([
+    408,
+    413,
+    429,
+    500,
+    502,
+    503,
+    504
+  ]);
+
+  async requestApiWithRetry (path: string, opts: RequestApiOptions, retryInterval?: number): Promise<any> {
+    const request = retryPromise<any>(
+      this.requestApi.bind(this, path, opts),
+      (error: any) => error && HttpClient.retryStatusCodes.has(error.status),
+      retryInterval
+    );
```

Comment:
See `retryPromise()` for the actual retry interval logic. Essentially, we want to keep retrying `this.requestApi()` until it resolves or rejects with an error other than a retriable status code.